runner/runners/polling_test.go

```diff
@@ -149,3 +148,77 @@ func TestPolling_ErrorPolling(t *testing.T) {
 	ex.Resume()
 
 }
+
+func TestPollingWorker_Intervalse(t *testing.T) {
+	_, _, poller := setupPoller()
+
+	st, err := poller.Run(&runner.Command{Argv: []string{"complete 42"}})
+	if err != nil {
+		t.Fatal(st, err)
+	}
+	st, _, err = runner.FinalStatus(poller, st.RunID)
+	if err != nil {
+		t.Fatal(st, err)
+	}
+	if st.State != runner.COMPLETE || st.ExitCode != 42 {
+		t.Fatal(st, err)
+	}
+}
+
+func setupTestPollerIntervals(period time.Duration) (*execers.SimExecer, runner.StatusQueryNower, runner.Service) {
+	ex := execers.NewSimExecer()
+	filerMap := runner.MakeRunTypeMap()
+	filerMap[runner.RunTypeScoot] = snapshot.FilerAndInitDoneCh{Filer: snapshots.MakeInvalidFiler(), IDC: nil}
+	single := NewSingleRunner(ex, filerMap, NewNullOutputCreator(), nil, stats.NopDirsMonitor, runner.EmptyID)
+	chaos := NewChaosRunner(single)
+	var nower runner.StatusQueryNower
+	nower = &instrumentedNower{ChaosRunner: *chaos}
+	poller := NewPollingService(chaos, nower, period)
+	return ex, nower, poller
+}
+
+type instrumentedNower struct {
+	ChaosRunner
```

Comment:
I think the current go vet error can be fixed by making this a *ChaosRunner and passing the pointer on line 175.

That being said I tried this out locally and it failed 50% of the time for me - trying to test using sleep on the microsecond scale is going to be extremely flaky due to cpu speed and contention. Strongly recommend if we are going to spend more than ~minutes getting a test to work reliably that we scrap it and move on, it is clear enough that the fix in polling.go is going to work.

Reply:
I reverted the test changes since it was also quite flaky for me.